### PR TITLE
Non-developer installation instructions

### PR DESCRIPTION
**Pull request by wusui**
Signed-off-by: Warren Usui <wusui@magna002.ceph.redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,68 @@
...
+### Install executables
+
+On the host machine on which you will run ansible-playbook, do the following:
+- sudo su -
+- mkdir ~/cephmetrics
+- subscription-manager repos --enable rhel-7-server-rhscon-2-installer-rpms
+- cd /etc/yum.repos.d/
+- curl -L -o cephmetrics-deps.repo https://chacra.ceph.com/repos/cephmetrics/dependencies/HEAD/rhel/7/repo
+- sed -i -e 's/\[cephmetrics/\[cephmetrics-deps/' cephmetrics-deps.repo
+- curl -L -o cephmetrics.repo https://shaman.ceph.com/api/repos/cephmetrics/master/latest/centos/7/repo/
```
Comment (zmc) [inline README.md]: The above three steps will be replaced by a single curl of a .repo file on download.ceph.com shortly

File: README.md [hunk 2]
```diff
@@ -1,2 +1,68 @@
...
+
+    [rgws]
+    rgw0.example.com
+
+Omit the mdss section if no ceph mds nodes are installed.  Omit the rgws section if no rgw nodes are installed.
+
+## Edit the vars.yml file
+
+Vars.yml is used to override defaults used by the ansible installation procedure.  Run the following:
+- echo "devel_mode: false" > ~/cephmetrics/vars.yml
```
Comment (zmc) [inline README.md]: This is actually the default in the packaged version, so we can omit this step

File: README.md (hunk 1 above)
Comment (zmc) [inline README.md]: http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo

Comment (zmc) [inline README.md]: so: `curl -L -O http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,68 @@
...
+- A host machine on which to execute `ansible-playbook` to orchestrate the deployment must be available.  In this document, this will be the same host as the dashboard host.
+- Passwordless SSH access from the deploy host to the ceph hosts. The username should be the same for all hosts.
+- Passwordless sudo access on the ceph and dashboard hosts
+- All hosts must share the same DNS domain
+
+## Installation
+
+### Install executables
+
+On the host machine on which you will run ansible-playbook, do the following:
```
Comment (zmc) [inline README.md]: We may want to wrap these steps in backticks `so they look like this`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,68 @@
...
+- A functional ceph cluster running version ceph-osd-10.2.7-27.el7cp.x86_64 or later is already up and running.
+- Another host machine independent of the ceph machines must be available.  This host will be used to receive data pushed by the hosts in the Ceph cluster, and will run the dashboard to display that data.
+- A host machine on which to execute `ansible-playbook` to orchestrate the deployment must be available.  In this document, this will be the same host as the dashboard host.
+- Passwordless SSH access from the deploy host to the ceph hosts. The username should be the same for all hosts.
+- Passwordless sudo access on the ceph and dashboard hosts
+- All hosts must share the same DNS domain
+
+## Installation
+
+### Install executables
```
Comment (zmc) [inline README.md]: We also need to inform the customer that they have to install [the repo file](http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo) on *all* the ceph cluster hosts.

**Comment by wusui** [comment]
I have pushed another set of changes to origin wip-metrics-wusui

**Comment by zmc** [comment]
@pcuzner, some responses:

* We can mention the version, sure, but it's worth noting that it's pulled in via dependencies.
* I don't see ceph-ansible being installed on the ceph nodes, in the docs or reality. Perhaps you meant 'on the dashboard host'? cephmetrics-ansible requires ceph-ansible for reasons I don't fully understand but I imagine @b-ranto wanted to merely pull in its dependencies.
* Good point, we'll do this
* We can also do this
* Good idea

1. That's a good question. Technically we could ship the repo out from the dashboard host, but I had been told we were to assume repos were already in place everywhere
2. These bits are in reality somewhere between "upstream" and "downstream" - they didn't go through our downstream process but are aimed at RHEL so are not exactly just upstream. I think after this week we can, if we want, work on making a CentOS-based workflow a bit easier and better-documented.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,84 @@
...
+```
+- sudo su -
+- cd /etc/yum.repos.d/
+- curl -L -O http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo
+- yum install cephmetrics-ansible
+```
+
+### Edit the inventory file
+
+A file named ~/cephmetrics/inventory needs to be created.  Ansible-playbook will use this inventory file when installing cephmetrics.  Inventory is an INI-like format file with entries for ceph-grafana and all the parts of the ceph cluster.  A template for this file can be copied from ~/cephmetrics/ansible/inventory.sample
```
Comment (zmc) [inline README.md]: The sample inventory has no way of being in the user's homedir. It is installed into `/usr/share/cephmetrics-ansible/inventory.sample`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,84 @@
...
+    grafana_host.example.com ansible_connection=local
+    ```
+Omit the mdss section if no ceph mds nodes are installed.  Omit the rgws section if no rgw nodes are installed.
+
+Ansible variables can be set in ~/cephmetrics/vars.yml if the user so desires.  [Click here](./ansible/ansible.md) for more information.
+
+## Run the ansible-playbook
+
+Run the following command:
+- ansible-playbook -v -i \~/cephmetrics/inventory -e '@\~/cephmetrics/vars.yml' playbook.yml
```
Comment (zmc) [inline README.md]: First, need to `cd /usr/share/cephmetrics-ansible`

File: README.md [hunk 7]
```diff
@@ -1,2 +1,84 @@
...
+    rgw0.example.com
+
+Since we are running `ansible-playbook` directly on the dashboard (`ceph-grafana`) host, the inventory entry should look like:
+    ```
+    [ceph-grafana]
+    grafana_host.example.com ansible_connection=local
+    ```
+Omit the mdss section if no ceph mds nodes are installed.  Omit the rgws section if no rgw nodes are installed.
+
+Ansible variables can be set in ~/cephmetrics/vars.yml if the user so desires.  [Click here](./ansible/ansible.md) for more information.
```
Comment (zmc) [inline README.md]: Broken link

File: README.md [hunk 8]
```diff
@@ -1,2 +1,84 @@
 # cephmetrics
-ceph metric collectors with collectd integration
+
+Cephmetrics is a tool that allows a user to visually monitor various metrics in a running Ceph cluster.
+
+## Disclaimer
+
+This document contains the installation procedure for cephmetrics released on June 30, 2017.
```
Comment (zmc) [inline README.md]: I think we should probably leave the above section out

File: README.md (hunk 5 above)
Comment (zmc) [inline README.md]: Don't capitalize names of binaries; could lead to confusion. Leave it lowercase and wrap with backticks like: `ansible-playbook`

File: README.md [hunk 9]
```diff
@@ -1,2 +1,84 @@
...
+- Another host machine independent of the ceph machines must be available.  This host will be used to receive data pushed by the hosts in the Ceph cluster, and will run the dashboard to display that data.
+- A host machine on which to execute `ansible-playbook` to orchestrate the deployment must be available.  In this document, this will be the same host as the dashboard host.
+- Passwordless SSH access from the deploy host to the ceph hosts. The username should be the same for all hosts.
+- Passwordless sudo access on the ceph and dashboard hosts
+- All hosts must share the same DNS domain
+
+## Expected configuration
+
+After running this procedure, the following configuration is expected.
+- The ceph nodes will basically remain unchanged, except that repos necessary for interacting with the cephmetrics host machine will be installed.
```
Comment (zmc) [inline README.md]: "The ceph nodes will have `collectd` installed, along with collector plugins from `cephmetrics-collectd`"

File: README.md [hunk 10]
```diff
@@ -1,2 +1,84 @@
...
+- The cephmetrics host machine will have a dashboard installed which the user can then use to observe the ceph network.
```
Comment (zmc) [inline README.md]: "The dashboard host will have `grafana` installed and configured to display various dashboards by querying data received from Ceph nodes via a `graphite-web`, `python-carbon` and `python-whisper` stack."

File: README.md [hunk 11]
```diff
@@ -1,2 +1,84 @@
...
+Since we are running `ansible-playbook` directly on the dashboard (`ceph-grafana`) host, the inventory entry should look like:
+    ```
+    [ceph-grafana]
+    grafana_host.example.com ansible_connection=local
+    ```
+Omit the mdss section if no ceph mds nodes are installed.  Omit the rgws section if no rgw nodes are installed.
+
+Ansible variables can be set in ~/cephmetrics/vars.yml if the user so desires.  [Click here](./ansible/ansible.md) for more information.
+
+## Run the ansible-playbook
```
Comment (zmc) [inline README.md]: "Deploy via ansible-playbook"

File: README.md [hunk 12]
```diff
@@ -1,2 +1,84 @@
...
+
+### Install executables
+
+On the host machine on which you will run ansible-playbook, do the following steps.  This will install a repo which includes the cephmetrics installation code and ansible (version 2.2.3 or later):
+```
+- sudo su -
+- mkdir ~/cephmetrics
+- subscription-manager repos --enable rhel-7-server-rhscon-2-installer-rpms
+- cd /etc/yum.repos.d/
+- curl -L -O http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo
```
Comment (wusui) [inline README.md]: Shouldn't that be:
url -L -o /etc/yum.repos.d/cephmetrics.repo -O http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo

Comment (wusui) [inline README.md]: -O missing from your line

File: README.md (hunk 8 above)
Comment (wusui) [inline README.md]: Done

File: README.md (hunk 9 above)
Comment (wusui) [inline README.md]: Done

File: README.md (hunk 10 above)
Comment (wusui) [inline README.md]: Done

File: README.md [hunk 13]
```diff
@@ -1,2 +1,84 @@
...
+- yum install cephmetrics-ansible
+```
+
+The cephmetrics repo also needs to be installed on all the ceph nodes as well.  Run the following steps on each ceph host:
+```
+- sudo su -
+- cd /etc/yum.repos.d/
+- curl -L -O http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo
```
Comment (wusui) [inline README.md]: also is -O needed here?

File: README.md [hunk 14]
```diff
@@ -18,34 +14,30 @@ This document contains the installation procedure for cephmetrics released on Ju
 ## Expected configuration
```
Comment (zmc) [inline README.md]: "Resulting configuration"

File: README.md [hunk 15]
```diff
@@ -18,34 +14,30 @@ This document contains the installation procedure for cephmetrics released on Ju
...
 
 After running this procedure, the following configuration is expected.
```
Comment (zmc) [inline README.md]: "After running this procedure, you will have the following configuration"

File: README.md [hunk 16]
```diff
@@ -18,34 +14,30 @@ This document contains the installation procedure for cephmetrics released on Ju
...
-- cd /etc/yum.repos.d/
-- curl -L -O http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo
-- yum install cephmetrics-ansible
+sudo curl -L -o /etc/yum.repos.d/cephmetrics.repo -O http://download.ceph.com/cephmetrics/rpm-master/el7/cephmetrics.repo
 ```
 
 ### Edit the inventory file
 
-A file named ~/cephmetrics/inventory needs to be created.  Ansible-playbook will use this inventory file when installing cephmetrics.  Inventory is an INI-like format file with entries for ceph-grafana and all the parts of the ceph cluster.  A template for this file can be copied from ~/cephmetrics/ansible/inventory.sample
+A file named ~/cephmetrics/inventory needs to be created.  `ansible-playbook` will use this inventory file when installing cephmetrics.  Inventory is an INI-like format file with entries for ceph-grafana and all the parts of the ceph cluster.  A template for this file can be copied from /usr/share/cephmetrics-ansible/inventory.sample
```
Comment (zmc) [inline README.md]: Wrap the path in backticks please

File: README.md [hunk 17]
```diff
@@ -75,10 +67,12 @@ Since we are running `ansible-playbook` directly on the dashboard (`ceph-grafana
...
+Ansible variables can be set in ~/cephmetrics/vars.yml if the user so desires.  [Click here](./ansible/README.md) for more information.
 
-Run the following command:
-- ansible-playbook -v -i \~/cephmetrics/inventory -e '@\~/cephmetrics/vars.yml' playbook.yml
+## Deploy via ansible-playbook
 
+Run the following commands:
+```
+cd /usr/share/cephmetrics-ansible
+ansible-playbook -v -i \~/cephmetrics/inventory -e '@\~/cephmetrics/vars.yml' playbook.yml
```
Comment (zmc) [inline README.md]: Remove the backslashes please

File: README.md (hunk 13 above)
Comment (zmc) [inline README.md]: No, `-o` is "write to this path" whereas `-O` is "use the filename provided by the remote server"

File: README.md (hunk 12 above)
Comment (zmc) [inline README.md]: No; see my other comment. `-o` and `-O` conflict.

**Pull request merged**